### PR TITLE
Fix migrating users whose username is the beginning of another username

### LINKED ISSUE #14: Migrating a user-id which begins like another one will fail
Reproduce:

- Create user "user1"
- Create user "user11"

migrate with the following config:

`user1:user22`

Traceback:

```
Traceback (innermost last):

Module ZPublisher.Publish, line 138, in publish
Module ZPublisher.mapply, line 77, in mapply
Module ZPublisher.Publish, line 48, in call_object
Module z3c.form.form, line 233, in __call__
Module z3c.form.form, line 228, in update
Module plone.app.z3cform.csrf, line 21, in execute
Module z3c.form.action, line 98, in execute
Module z3c.form.button, line 315, in __call__
Module z3c.form.button, line 170, in __call__
Module ftw.usermigration.browser.migration, line 308, in handleMigrate
Module ftw.usermigration.users, line 24, in migrate_users
KeyError: 'user1'
```

Analyse:

The `searchUsers` (https://github.com/4teamwork/ftw.usermigration/blob/master/ftw/usermigration/users.py#L16) method seems to do a wildcard-search

### PR DESCRIPTION
**Pull request by elioschmutz**
See #14 for more information.

Closes #14

### CODE REVIEW

File: ftw/usermigration/tests/test_users.py [hunk 1]
```diff
@@ -116,3 +117,21 @@ def test_delete_users(self):
             (),
             uf.searchUsers(id='john.deo')
         )
+
+    def test_migrate_user_which_is_a_part_of_another_user_does_not_fail(self):
```
Comment (lukasgraf) [inline ftw/usermigration/tests/test_users.py]: `test_migrate_userid_which_is_a_substring_of_another_userid`?

File: setup.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
     'ftw.testing',
     'ftw.builder',
     'ftw.testbrowser',
+    'plone.api'
```
Comment (lukasgraf) [inline setup.py]: Please add a trailing comma after `'plone.api'`

File: ftw/usermigration/users.py [hunk 3]
```diff
@@ -13,42 +14,40 @@ def migrate_users(context, mapping, mode='move', replace=False):
...
-
-                    if mode == 'move':
-                        moved.append(('acl_users', old_userid, new_userid))
-                    if mode == 'copy':
-                        copied.append(('acl_users', old_userid, new_userid))
-                    if mode == 'delete':
-                        deleted.append(('acl_users', old_userid, None))
+        for plugin_id, plugin in uf.plugins.listPlugins(IUserEnumerationPlugin):
+            # Only ZODB User Manager is supported
+            if not IZODBUserManager.providedBy(plugin):
```
Comment (lukasgraf) [inline ftw/usermigration/users.py:19]: Please don't sneak refactorings into functional parts of the PR without mentioning them. Ideally, do them in a separate commit.

**Review by lukasgraf — CHANGES_REQUESTED** [review]
Some minor points, looks good to me otherwise 👍

**Comment by elioschmutz** [comment]
@lukasgraf 👍  done! I also split the commit into a functional and a refactoring commit.

**Review by lukasgraf — APPROVED** [review]

**Comment by buchi** [comment]
@elioschmutz looks good to me. Can you fix conflicts in `HISTORY.txt` and rebase?

**Comment by elioschmutz** [comment]
@buchi 👍

File: docs/HISTORY.txt [hunk 4]
```diff
@@ -4,8 +4,8 @@ Changelog
 1.2 (unreleased)
 ----------------
 
+- Fix migrating users whose username is the beginning of another username [elioschmutz]
 - Implement group members migration [elioschmutz]
```
Comment (buchi) [inline docs/HISTORY.txt]: @elioschmutz Can you add newlines between changelog entries? Should be consistent with the previous ones.

**Comment by elioschmutz** [comment]
@buchi 👍

**Pull request merged**